### PR TITLE
Fix 404 when click compare on non-forked repository

### LINKED ISSUE #936: Compare-Button leads to 404
- Gitea version (or commit ref): 1.0.0+272-gcf47532e
- Git version: 2.8.3
- Operating system: Linux ad0c0ea10d83 4.4.0-62-generic #83-Ubuntu
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL): https://try.gitea.io/gitea/gitea/compare/master...gitea:master
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
When creating a repository, next to the branch-dropdown, it shows an image to compare branches. That link is always 404, since it will be of the form 
`try.gitea.io/<user>/<repo>/compare/<branch1>...<user>:<branch2>`
It should be:
`try.gitea.io/<user>/<repo>/compare/<branch1>...<branch2>`
If I change the address in the bar and remove the second user reference, the site loads fine.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/42030136-compare-button-leads-to-404?utm_campaign=plugin&utm_content=tracker%2F47456670&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F47456670&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by lunny**
Fix #936

### CODE REVIEW

**Comment by Bwko** [comment]
This fix will reintroduce #304
You could cherry pick [this commit](https://github.com/gogits/gogs/commit/e43479d948b1e3a71dd4488ee1419e45017ff5c1)

**Comment by lunny** [comment]
OK. I will see that.

**Comment by lunny** [comment]
@Bwko done.

**Comment by Bwko** [comment]
LGTM

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -10,7 +10,7 @@
 		<div class="ui secondary menu">
 			{{if .PullRequestCtx.Allowed}}
 				<div class="fitted item">
-					<a href="{{.BaseRepo.Link}}/compare/{{.BaseRepo.DefaultBranch}}...{{.Username}}:{{.BranchName}}">
+					<a href="{{.BaseRepo.Link}}/compare/{{.BaseRepo.DefaultBranch}}...{{.BranchName}}">
```
Comment (ethantkoenig) [inline templates/repo/home.tmpl:13]: Removing `{{.Username}}` fixes two-branches-inside-the-same-repo comparisons, but break comparisons across forks.

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: templates/repo/home.tmpl (hunk 1 above)
Comment (lunny) [reply templates/repo/home.tmpl:13]: But on compare UI, it's not ready for compare between forks. I think this need another PR to do that. For this one, we should let the button works at first.
Gitea's vs Github's
untitled

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

File: templates/repo/home.tmpl (hunk 1 above)
Comment (strk) [reply templates/repo/home.tmpl:13]: How about you just make sure that `.Username` expands to what you expect it to expand, and *later* work on allowing different names ?

Comment (lunny) [reply templates/repo/home.tmpl:13]: I think the later work should be similar with github. Add two dropdown menus and list the forked repos.

**Comment by lunny** [comment]
I will close this one since it will break comparing across forks and I will send another one.

**Comment by lunny** [comment]
Deprecated by #1104

**Pull request closed without merging**